### DISCUSSION TITLE
Republishing GitHub page

### DISCUSSION THREAD

**Discussion by DharaniSureshBabu** [cu-esiil-edu/2024-data-short-course#3] [Q&A]
@eculler @nquarder How to publish again? once I unpublish it. Thank you.

**Comment by nquarder** [comment]
Hi @DharaniSureshBabu - it looks like you were able to get your page republished (https://dharanisureshbabu.github.io/). Great job! It looks really good, and you managed to get a lot accomplished after class yesterday. If you have time, could you add the steps you followed to get your page re-published after un-publishing? Again, great job with this personal portfolio. -NQ

**Reply by DharaniSureshBabu** [reply]
@nquarder @eculler To re-publish my GitHub Pages site after I'd previously unpublished it, I followed these steps:

1. **Log in to GitHub**: Open your web browser, go to GitHub.com, and log in with your username and password.
2. **Navigate to Your Repository**: Go to the repository that is linked to your GitHub Pages site. You can find your repositories by clicking on your profile icon in the top right corner and selecting "Your repositories".
3. **Access the Settings**: Once you're in your repository, look for the "Settings" tab near the top of the page and click on it.
4. **Go to the Pages Section**: In the Settings menu, scroll down the sidebar until you find the "Pages" section. Click on it to open the GitHub Pages settings.
5. **Choose a Source**: In the "Source" section, you’ll see a dropdown menu. Click on it and select the branch you want to publish your site from (commonly the `main` or `master` branch) and the folder (typically `/ (root)` or `/docs`). **This action re-establishes the publishing source for your GitHub Pages.**
6. **Save Your Changes**: After selecting your publishing source, click the "Save" button to apply your changes.
7. **Customize Your Site (Optional)**: If you want, you can customize your site's domain, enforce HTTPS, and make other changes in the same "Pages" settings section.
8. **Wait for the Publication**: Once you save your changes, GitHub will start building your site. It can take a few minutes for your site to be published. You’ll see a link to your published site at the top of the "Pages" settings section when it's ready.
9. **Verify Your Site is Live**: Click the provided link to your site to ensure it is live. If you've made significant changes or are re-publishing, it might take a few minutes for all changes to appear due to caching.

Remember, if you have custom domains or specific build configurations, you might need to revisit those settings to ensure everything works as expected after re-publishing.
